ReactiveCocoa/Swift/Signal.swift

```diff
@@ -140,7 +140,29 @@ public final class Signal<T, E: ErrorType> {
 			return nil
 		}
 	}
+}
+
+public protocol SignalType {
+	/// The type of values being sent on the signal.
+	typealias T
+	/// The type of error that can occur on the signal. If errors aren't possible
+	/// then `NoError` can be used.
+	typealias E: ErrorType
+
+	/// Extracts a signal from the receiver.
+	var signal: Signal<T, E> { get }
```

Comment:
Turns out there are a few places relying on this that would otherwise require creating a new signal simply to `observe` the original.